src/python/tests/storage/test_storage_api.py

```diff
@@ -0,0 +1,283 @@
+        assert result["rows"][0]["key"] == "TEST_1"
+        assert result["rows"][0]["description"] == "Testing the storage service data 1"
+        assert result["rows"][0]["value"] == 10
+        assert result["rows"][0]["previous_value"] == 2
+
+        assert result["rows"][1]["key"] == "TEST_2"
+        assert result["rows"][1]["description"] == "Testing the storage service data 2"
+        assert result["rows"][1]["value"] == 15
+        assert result["rows"][1]["previous_value"] == 2
+
+    def test_where_query_param(self):
+        result = Storage().query_tbl("statistics", PayloadBuilder().WHERE(["key", "=", "TEST_1"]).query_params())
+        assert len(result["rows"]) == 1
+        assert result["count"] == 1
+        assert result["rows"][0]["key"] == "TEST_1"
+        assert result["rows"][0]["description"] == "Testing the storage service data 1"
+        assert result["rows"][0]["value"] == 10
+        assert result["rows"][0]["previous_value"] == 2
+
+    def test_where_payload(self):
+        result = Storage().query_tbl_with_payload("statistics", PayloadBuilder().WHERE(["value", "!=", 15]).payload())
+        assert len(result["rows"]) == 1
+        assert result["count"] == 1
+        assert result["rows"][0]["key"] == "TEST_1"
+        assert result["rows"][0]["description"] == "Testing the storage service data 1"
+        assert result["rows"][0]["value"] == 10
+        assert result["rows"][0]["previous_value"] == 2
+
+    @pytest.mark.skip(reason="FOGL-615")
+    def test_where_invalid_key(self):
+        result = Storage().query_tbl_with_payload("statistics", PayloadBuilder().
+                                                  WHERE(["bla", "=", "invalid"]).payload())
+        assert "ERROR" in result["message"]
+
+    @pytest.mark.skip(reason="FOGL-607 #7")
+    def test_multiple_and_where(self):
+        result = Storage().query_tbl_with_payload("statistics", PayloadBuilder().
+                                                  WHERE(["value", "!=", 0]).
+                                                  AND_WHERE(["key", "=", "TEST_1"]).
+                                                  AND_WHERE(["previous_value", "<", 10]).payload())
+        assert len(result["rows"]) == 1
+        assert result["count"] == 1
+        assert result["rows"][0]["key"] == "TEST_1"
+        assert result["rows"][0]["description"] == "Testing the storage service data 1"
+        assert result["rows"][0]["value"] == 10
+        assert result["rows"][0]["previous_value"] == 2
+
+    @pytest.mark.skip(reason="FOGL-607 #7")
+    def test_multiple_or_where(self):
+        result = Storage().query_tbl_with_payload("statistics", PayloadBuilder().
+                                                  WHERE(["value", "=", 0]).
+                                                  OR_WHERE(["key", "=", "TEST_1"]).
+                                                  OR_WHERE(["previous_value", ">", 10]).payload())
+        assert len(result["rows"]) == 1
```

Comment:
should not the test data be like that AND has 1 result and OR gives 2 results, while having atleast 3 records in table